### PR TITLE
Fix: update to wsproto 0.13 for real this time

### PR DESCRIPTION
**Pull request by RPigott**
A proper update for wsproto 0.13

### CODE REVIEW

**Comment by RPigott** [comment]
It looks like the travis build is affected by https://github.com/pypa/pip/issues/6163, maybe pass --no-use-pep517 for PyInstaller until it is fixed in pip?

File: mitmproxy/proxy/protocol/websocket.py [hunk 1]
```diff
@@ -52,51 +53,52 @@ def __init__(self, ctx, handshake_flow):
...
+                server_extensions = [PerMessageDeflate()]
+        self.connections[self.client_conn] = WSConnection(ConnectionType.SERVER)
+        self.connections[self.server_conn] = WSConnection(ConnectionType.CLIENT)
+
+        if client_extensions:
+            client_extensions[0].finalize(handshake_flow.response.headers['Sec-WebSocket-Extensions'])
+        if server_extensions:
+            server_extensions[0].finalize(handshake_flow.response.headers['Sec-WebSocket-Extensions'])
+
+        request = Request(extensions = client_extensions, host = handshake_flow.request.host, target = handshake_flow.request.path)
```
Comment (Kriechi) [inline mitmproxy/proxy/protocol/websocket.py]: Please remove the whitespace around the `=` inside a function call (args)

File: mitmproxy/proxy/protocol/websocket.py [hunk 2]
```diff
@@ -127,51 +129,52 @@ def get_chunk(payload):
...
 
         return False
 
     def _inject_messages(self, endpoint, message_queue):
         while True:
             try:
                 payload = message_queue.get_nowait()
-                self.connections[endpoint].send_data(payload, final=True)
-                data = self.connections[endpoint].bytes_to_send()
+                data = self.connections[endpoint].send(Message(data = payload, message_finished = True))
```
Comment (Kriechi) [inline mitmproxy/proxy/protocol/websocket.py]: same whitespace as above

**Comment by Kriechi** [comment]
@RPigott would you mind opening a new issue / PR for the pyinstaller problem?

**Pull request merged**